nodebuilder/share/module.go

```diff
@@ -47,6 +48,17 @@ func ConstructModule(tp node.Type, cfg *Config, options ...fx.Option) fx.Option
 		return fx.Module(
 			"share",
 			baseComponents,
+			fx.Provide(fx.Annotate(
+				func(path node.ConfigPath, ds datastore.Batching) (*eds.Store, error) {
+					return eds.NewStore(string(path), ds)
+				},
+				fx.OnStart(func(ctx context.Context, eds *eds.Store) error {
+					return eds.Start(ctx)
+				}),
+				fx.OnStop(func(ctx context.Context, eds *eds.Store) error {
+					return eds.Stop(ctx)
+				}),
```

Comment:
It might be the case that we are using fx version that does not forces us to pass ctx to lifecycle funcs

Reply:
I don't see anything wrong with it though? We are doing this everywhere else

Reply:
Nothing wrong, just a simplification suggestion